solanches/tests/test_controller.py

```diff
@@ -148,3 +147,90 @@ def test_remove_comercio_sucesso(mock_remove_comercio, mock_get_by_name,  contro
     mock_remove_comercio.return_value = 1
     result = controller.remove_comercio(comercio_nome)
     assert result == 1
+
+
+@pytest.mark.skip(reason="TODO")
+@mock.patch('solanches.models.Comercio.get_by_name')
+@mock.patch('solanches.models.Cardapio.get_by_id')
+def test_cadastra_comercio(mock_get_by_name, mock_get_by_id, controller):
+    comercio_nome = 'comercio_test1'
+    comercio_attributes = {
+            'telefone': '12234'
+    }
+    result = controller.cadastra_comercio(comercio_nome, comercio_attributes)
+    mock_get_by_id.return_value = {'produtos': []}
+    mock_get_by_name.return_value = {'nome': 'comercio_teste1', 'attributes': {"telefone": "12234"}}
+
+    expected_fields = ["nome", "attributes", "created_at"]
+    result_fields = result.keys()
+    assert all(field in result_fields for field in expected_fields)
+
+
+def test_cadastra_comercio_sem_nome(controller):
+    comercio_nome = 90992727
+    password = "849439030"
+    comercio_attributes = {
+        "endereco": "ruaa",
+        "horarios": "21h-24h"
+    }
+    with pytest.raises(SolanchesBadRequestError) as exinfo:
+        controller.cadastra_comercio(comercio_nome, password,  comercio_attributes)
+    assert str(exinfo.value.message) == 'Erro: campo nome inválido!'
+
+
+def test_cadastra_comercio_sem_senha(controller):
+    comercio_nome = "comercio1"
+    password = 0
+    comercio_attributes = {
+        "endereco": "ruaa",
+        "horarios": "21h-24h"
+    }
+    with pytest.raises(SolanchesBadRequestError) as exinfo:
+        controller.cadastra_comercio(comercio_nome, password,  comercio_attributes)
+    assert str(exinfo.value.message) == "Erro: Senha não informada!"
+
+
+def test_cadastra_comercio_atributos_invalidos(controller):
+    comercio_nome = 'comercio_test4'
+    password = "873838383"
+    comercio_attributes = 48488448
```

Comment:
Para deixar o teste mais legível
```suggestion
    attributes_nao_eh_dict = 48488448
```